File: PH2007SDK/developpers/CSTut1/myPlayer.cs
/* Tutorial done by
 * 
 * Richard Clark
 * Project Hoshimi Lead Manager
 * Contact at ori@c2i.fr
 * 
*/
using System;
using System.Collections.Generic;
using System.Text;
using PH.Common;
using PH.Map;
using System.Drawing;

namespace CSTut1
{
    public class myPlayer : PH.Common.Player
    {
        public myPlayer() { }
        public myPlayer(string name, int ID): base(name, ID)
        {
            this.ChooseInjectionPoint += new PH.Common.ChooseInjectionPointHandler(myPlayer_ChooseInjectionPoint);
            this.WhatToDoNext += new PH.Common.WhatToDoNextHandler(myPlayer_WhatToDoNext);
        }

        bool bCollectorBuilded;
        private void myPlayer_WhatToDoNext()
        {
            if (this.AI.State == NanoBotState.WaitingOrders)
            {
                if (!bCollectorBuilded)
                {
                    if (this.AI.Build(typeof(Collector)))
                        bCollectorBuilded = true;
                }
                else
                {
                    this.AI.MoveTo(new Point(100, 50));
                }
            }
        }

        private void myPlayer_ChooseInjectionPoint()
        {
            Entity ent = this.Tissue.get_EntitiesByType(EntityEnum.AZN)[0];
            this.InjectionPointWanted = new Point(120,50);
        }

        public override System.Drawing.Bitmap Flag
        {
            get{return Properties.Resources.rcFlag;}
        }
    }
}

File: PH2007SDK/developpers/VBTut1/myPlayer.vb
'Tutorial done by
' 
'Richard Clark
'Project Hoshimi Lead Manager
'Contact at ori@c2i.fr
' 
Imports PH.Common
Imports PH.Map
Imports System.Drawing

Public Class myPlayer
    Inherits PH.Common.Player

    Public Sub New()
    End Sub

    Public Sub New(ByVal name As String, ByVal ID As Integer)
        MyBase.New(name, ID)
    End Sub

    Private Sub myPlayer_ChooseInjectionPointEvent() Handles Me.ChooseInjectionPoint
        Me.InjectionPointWanted = New Point(140, 80)
    End Sub

    Dim bCollectorBuilded As Boolean
    Private Sub myPlayer_WhatToDoNextEvent() Handles Me.WhatToDoNext
        If Me.AI.State = NanoBotState.WaitingOrders Then
            If Not bCollectorBuilded Then
                If Me.AI.Build(GetType(Collector)) Then
                    bCollectorBuilded = True
                End If
            Else
                Me.AI.MoveTo(New Point(100, 50))
            End If
        End If
    End Sub

    Public Overrides ReadOnly Property Flag() As System.Drawing.Bitmap
        Get
            Return My.Resources.rcFlag
        End Get
    End Property

End Class



